location/src/commonMain/kotlin/location/LocationStateRepo.kt

```diff
@@ -132,13 +137,12 @@ open class LocationStateImplRepo(
 class LocationStateRepo(
     settingsBuilder: (CoroutineContext) -> BaseLocationManager.Settings,
     builder: BaseLocationManager.Builder,
-    coroutineContext: CoroutineContext = Dispatchers.Main.immediate,
-    contextCreator: CoroutineContext.(String) -> CoroutineContext = { this + singleThreadDispatcher(it) },
+    coroutineContext: CoroutineContext = Dispatchers.Main.immediate, // singleThreadDispatcher(it)
```

Comment:
Yeah which is it, @Daeda88 ?

Reply:
I would say it should be a fixed singleThreadDispatcher like in other similar pieces of code. I guess the Dispatchers.Main.immediate was overlooked by me, should have just been without a default scope originally.

Reply:
yeah so now a `lazy` declared top level `singleThreadDispatcher` I'd say.